### ISSUE TITLE
Distinct version support for create/upgrade/scale

### ISSUE THREAD

**Issue by jackfrancis** [Azure/acs-engine#3399]
Currently there is some implementation overloading w/ respect to version support across create, upgrade, and scale operations.

From a high level:

- simplify creation UX by reducing available versions to a reasonable subset (say, the latest 3 patch versions for a given minor release channel)
- retain full historical version "support" for upgrade and scale
  - i.e., if we "remove support for creating a version A.B.C cluster" we need to "maintain support for upgrading a version A.B.C cluster to A.B.D", where A.B.D is a "supported" version
  - i.e., if we "remove support for creating a version A.B.C cluster" we need to "maintain support for scaling up/down that A.B.C cluster"; we should not require a cluster to be upgraded in order to be scaleable

**Comment by CecileRobertMichon** [comment]
If version X and Y are deprecated, should we allow upgrade from version X to version Y? The risk with answering no to that question is that a cluster might be non-upgradable if the user is on version 1.7.7 and 1.7 and 1.8 have been deprecated since they can't make the jump to 1.8... Also makes the implementation more difficult because we would have to distinguish between "version the user is upgrading from" and "version the user is upgrading to" and "this is a scale" (we still want to allow scaling a cluster if the version is deprecated without requiring an upgrade). Right now the approach I'm taking is: Is this a cluster create? If so --> only allow a restricted number of k8s version that are currently supported. If not (ie. this is an update to an existing cluster, upgrade or scale) --> allow all versions that are valid and have once been supported.


TLDR; Should we allow upgrading to a deprecated version? I think we shouldn't but it's a lot easier if we do.

**Comment by jackfrancis** [comment]
Let's punt on answering that question. We'll deprecate 1.7.x for new creates, but we'll plan on keeping 1.8.x "supported" indefinitely so that 1.7 folks can upgrade to 1.8 (and 1.9, etc).

And frankly, once we deprecate 1.8. we can just cut a new version and announce that older clusters still running 1.7 will need to use the older acs-engine version to get to 1.8.
